### ISSUE TITLE
Define centroid for Point

### ISSUE THREAD

**Issue by juliohm** [JuliaGeometry/Meshes.jl#64]
In various algorithms there is a pattern `coordinates(centroid(domain, i))` that works well when the domain is a `Mesh`. However, when the domain is a `PointSet` the centroid isn't defined. We could simply add a method `centroid(p::Point) = p` to avoid over-engineering downstream. I am still thinking about the implications of doing this, but it sounds reasonable mathematically speaking?

Let me know if you feel that this doesn't make sense. I plan to add the method over the week as I find the time.

**Comment by serenity4** [comment]
I believe the centroid is defined as being the point that has the minimum distance with all other points, so in the case there is only one point, it is logically the point itself. I think it is reasonable.

**Comment by serenity4** [comment]
Actually, it is rigorously the arithmetic mean of all other points (from [Wikipedia](https://en.wikipedia.org/wiki/Centroid)), so it holds.

**Comment by juliohm** [comment]
I realized that in the latest version of the project the code `centroid(domain, i)` already works even for point sets because we specialized the `centroid` for the `PointSet` itself. Given that this already works we can save ourselves from adding a trivial `centroid(p) = p` for points for now.

We can reopen this issue in the future if we feel that we should add a trivial method like this. The premise I raised in the beginning of the issue is not valid with the latest release. :)

Thanks for the awesome feedback as usual. Always nice to have a mathematician working together on precise definitions 💯
